File: 프로그래머스/LV2/할인 행사.py
def solution(want, number, discount):
    answer = 0

    for i in range(0, len(discount) - 9):
        check = True
        for j in range(len(want)):
            if discount[i : i + 10].count(want[j]) < number[j]:
                check = False
        if check:
            answer += 1
    return answer

File: 프로그래머스/javascript/LV2/할인행사.js
function solution(want, number, discount) {
  let answer = 0; // 날짜 수를 카운트

  for (let i = 0; i <= discount.length - 10; i++) {
    const check = discount.slice(i, i + 10);
    let flag = false;
    for (let j = 0; j < want.length; j++) {
      if (check.filter((k) => k === want[j]).length !== number[j]) {
        flag = true;
        break;
      }
    }
    if (!flag) {
      answer++;
    }
  }

  return answer;
}
